NetworkPkg/NvmeOfDxe/NvmeOfCliInterface.c

```diff
@@ -78,20 +78,28 @@ NvmeOfCliDeleteMapEntries (
   )
 {
   LIST_ENTRY               *Entry;
+  LIST_ENTRY               *NextEntryProcessed = NULL;
   NVMEOF_CLI_CTRL_MAPPING  *MappingList;
   BOOLEAN                  Flag = FALSE;
+  VOID                     *Private = NULL;
 
-  NET_LIST_FOR_EACH (Entry, &gCliCtrlMap->CliCtrlrList) {
+  NET_LIST_FOR_EACH_SAFE (Entry, NextEntryProcessed, &gCliCtrlMap->CliCtrlrList) {
     MappingList = NET_LIST_USER_STRUCT (Entry, NVMEOF_CLI_CTRL_MAPPING, CliCtrlrList);
     if (MappingList->Ctrlr == Ctrlr) {
+      if (MappingList->Private != Private) {
```

Comment:
Please explain this comparison.

You seem to want to make sure that you don't free a MappingList->Privat pointer more than once. But it isn't safe to do this this way. What if the MappingList->Private pointer has values A, B, and A again while you traverse the list?

Reply:
Yes, MappingList entries are created one per each namespace of a controller in NvmeOfAttachCliCallback(). Ctrlr and Private are common across all the entries saved during one execution of NvmeOfAttachCliCallback. All such entries in the list will be one after the other i.e., values like A, B and A again are not possible. Private can differ in subsequent entries of the list only if they are namespaces from another connect command using CLI. Connecting to the same controller again is not allowed using CLI.

Reply:
> values like A, B and A again are not possible

It's hard for me to verify this, so I believe you. But I suggest you add a comment explaining the intention of the code, and why it works the way it's intended to work.